cpp/oneapi/dal/backend/primitives/lapack/test/syevd_dpc.cpp

```diff
@@ -128,6 +128,46 @@ class syevd_test : public te::float_algo_fixture<Float> {
         }
     }
 
+    void check_eigvals_with_eigen(const la::matrix<Float>& s,
+                                  const la::matrix<Float>& eigvecs,
+                                  const la::matrix<Float>& eigvals) const {
+        INFO("convert results to float64");
+        const auto s_f64 = la::astype<double>(s);
```

Comment:
As an alternative, perhaps the input values could be hard-coded along with the solutions instead of checking them against a different library.

Reply:
Yes, but with this solution there is no opportunity to use random generated data(I mean extend tests with for example row_count  = GENERATE(3, 28, 125, 256);) and also it will be complicated to check results for big datasets

Reply:
How about putting them in the existing folders with .csv files that have data and expected results?

Reply:
It's possible, but let's say for pca it's necessary to contain gold eigenvectors, eigenvalues, and it will increase the total size of the repo, especially with big datasets. I see no reasons to avoid this pr tbh